mods/fire/init.lua

```diff
@@ -109,27 +54,46 @@ minetest.register_node("fire:permanent_flame", {
 	sunlight_propagates = true,
 	floodable = true,
 	damage_per_second = 4,
-	groups = {igniter = 2, dig_immediate = 3},
+	groups = {igniter = 2, dig_immediate = 3, fire = 1},
 	drop = "",
+	on_flood = flood_flame
+}
+
+-- Permanent flame node
+local permanent_fire_node = table.copy(fire_node)
+permanent_fire_node.description = S("Permanent Flame")
+permanent_fire_node.groups.not_in_creative_inventory = 1
```

Comment:
This is the wrong way around.
The permanent flame should be in the creative inventory, the "basic flame" shouldn't be.